plugins/datastreamer_plugin/utils/dynamic_types_utils.cpp

```diff
@@ -0,0 +1,218 @@
+    {
+        eprosima::fastrtps::types::TypeKind kind = members.second->get_descriptor()->get_kind();
+
+        switch (kind)
+        {
+            case eprosima::fastrtps::types::TK_BOOLEAN:
+            case eprosima::fastrtps::types::TK_BYTE:
+            case eprosima::fastrtps::types::TK_INT16:
+            case eprosima::fastrtps::types::TK_INT32:
+            case eprosima::fastrtps::types::TK_INT64:
+            case eprosima::fastrtps::types::TK_UINT16:
+            case eprosima::fastrtps::types::TK_UINT32:
+            case eprosima::fastrtps::types::TK_UINT64:
+            case eprosima::fastrtps::types::TK_FLOAT32:
+            case eprosima::fastrtps::types::TK_FLOAT64:
+            case eprosima::fastrtps::types::TK_FLOAT128:
+                // Numeric case
+                numeric_type_names.push_back({base_type_name + separator + members.first,
+                                              members.second->get_id(), kind});
+                break;
+
+            case eprosima::fastrtps::types::TK_CHAR8:
+            case eprosima::fastrtps::types::TK_CHAR16:
+            case eprosima::fastrtps::types::TK_STRING8:
+            case eprosima::fastrtps::types::TK_STRING16:
+            case eprosima::fastrtps::types::TK_ENUM:
+                // String case
+                string_type_names.push_back({base_type_name + separator + members.first, members.second->get_id(),
+                                             kind});
+                break;
+
+            case eprosima::fastrtps::types::TK_NONE:
+            case eprosima::fastrtps::types::TK_ALIAS:
+            case eprosima::fastrtps::types::TK_BITMASK:
+            case eprosima::fastrtps::types::TK_ANNOTATION:
+            case eprosima::fastrtps::types::TK_STRUCTURE:
+            case eprosima::fastrtps::types::TK_UNION:
+            case eprosima::fastrtps::types::TK_BITSET:
+            case eprosima::fastrtps::types::TK_SEQUENCE:
+            case eprosima::fastrtps::types::TK_ARRAY:
+            case eprosima::fastrtps::types::TK_MAP:
+                // Complex types
+                // TODO
+                break;
+
+            default:
+                // Shouold not happen
+                break;
+        }
+    }
+}
+
+void get_introspection_data(
+        eprosima::fastrtps::types::DynamicType_ptr type,
+        const TypeIntrospectionStruct& numeric_type_names,
+        const TypeIntrospectionStruct& string_type_names,
+        eprosima::fastrtps::types::DynamicData_ptr data,
+        TypeIntrospectionNumericData& numeric_data,
+        TypeIntrospectionStringData& string_data)
+{
+    // First get numeric data
+    // It is used by index of the vector to avoid unnecessary lookups, but vectors must be sorted
+    for (int i = 0; i < numeric_type_names.size(); ++i)
+    {
+        numeric_data[i].second = get_numeric_type_from_data(
+            data, std::get<1>(numeric_type_names[i]),
+            std::get<2>(numeric_type_names[i]));
```

Comment:
This must be `std::get<0>` and `std::get<1>` (won't even compile otherwise)

Reply:
Execptions will be handled in future PR